plugins/s3_auth/unit_tests/test_aws_auth_v4.cc

```diff
@@ -69,37 +69,59 @@ TEST_CASE("uriEncode(): encode reserved chars in an object name", "[AWS][auth][u
   CHECK_FALSE(encoded.compare("%20/%21%22%23%24%25%26%27%28%29%2A%2B%2C%3A%3B%3C%3D%3E%3F%40%5B%5C%5D%5E%60%7B%7C%7D"));
 }
 
-TEST_CASE("uriDecode(): decode empty input", "[AWS][auth][utility]")
+TEST_CASE("isUriEncoded(): check an empty input", "[AWS][auth][utility]")
 {
-  String encoded("");
-  String decoded = uriDecode(encoded);
-  CHECK(0 == decoded.length()); /* 0 encoded because of the invalid input */
+  CHECK(false == isUriEncoded(""));
 }
 
-TEST_CASE("uriDecode(): decode unreserved chars", "[AWS][auth][utility]")
+TEST_CASE("isUriEncoded(): '%' and nothing else", "[AWS][auth][utility]")
+{
+  CHECK(false == isUriEncoded("%"));
+}
+
+TEST_CASE("isUriEncoded(): '%' but no hex digits", "[AWS][auth][utility]")
+{
+  CHECK(false == isUriEncoded("XXX%XXX"));
+}
+
+TEST_CASE("isUriEncoded(): '%' but only one hex digit", "[AWS][auth][utility]")
+{
+  CHECK(false == isUriEncoded("XXXXX%2XXXXXX"));
+}
+
+TEST_CASE("isUriEncoded(): '%' and 2 hex digit", "[AWS][auth][utility]")
+{
+  CHECK(true == isUriEncoded("XXX%12XXX"));
+}
+
+TEST_CASE("isUriEncoded(): space not encoded", "[AWS][auth][utility]")
+{
+  CHECK(false == isUriEncoded("XXXXX XXXXXX"));
+}
+
+TEST_CASE("isUriEncoded(): object name and '/'", "[AWS][auth][utility]")
+{
+  CHECK(false == isUriEncoded("XXXXX/XXXXXX", /* isObjectName */ false));
+  CHECK(true == isUriEncoded("XXXXX%2FXXXXXX", /* isObjectName */ false));
+  CHECK(false == isUriEncoded("XXXXX/XXXXXX", /* isObjectName */ true));
+  CHECK(true == isUriEncoded("XXXXX%2FXXXXXX", /* isObjectName */ true));
```

Comment:
* Yes, it is true that `isObjectName=true` is not used in the plugin code, I was torn but kept it mostly because the initial `UriEncode()` implementation followed strictly [AWS specific encoding spec](http://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html) which supports object names differently. 
* Added a couple of extra tests similar to the one you suggested with some comments.

Reply:
> Yes, it is true that `isObjectName=true` is not used in the plugin code, I was torn but kept it mostly because the initial `UriEncode()` implementation followed strictly [AWS specific encoding spec](http://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html) which supports object names differently.

OK sounds good.

> Added a couple of extra tests similar to the one you suggested with some comments.

This sounds good but for some reason I do not see the test case I mentioned. Can you give a line number?

Reply:
haven't pushed the code yet, coming ...

Reply:
Nice.